### ISSUE TITLE
Verification failing with "cannot read property of 'admins' of undefined"

### ISSUE THREAD

**Issue by masau** [clems4ever/authelia#126]
I get this error when authelia tries to verify after successfully authenticating.
My ldap server is freeipa running on centos. Authelia is from npm and is running on arch linux along w/ the nginx server. "admins" in the error is definitely referring to the ldap group - when I take my user out of the admins group I get the same error with "server_admins" instead.

Error w/ context:
```
info: Request GET handled on /verify                                                                                                                                                          
debug: Verify: headers are {"x-original-uri":"/","x-real-ip":"172.16.41.30","host":"authelia.mydomain.com","connection":"close","user-agent":"Mozilla/5.0 (X11; Linux x86_64; rv:55.0) Gecko/201
00101 Firefox/55.0","accept":"text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8","accept-language":"en-US,en;q=0.5","accept-encoding":"gzip, deflate, br","cookie":"_ga=GA1.2.10
89724715.1482724523; connect.sid=s%3A2D893_odUMDjihoMoXsAhTEKMvcHHtQl.Gm7tJIdU6b82wiTu1MMtpHTAtzbHm4Lrt5YuCED2hww","dnt":"1","upgrade-insecure-requests":"1"}                                 
debug: Verify: only_basic_auth=false                                                                                                                                                          
debug: Verify: domain=authelia.mydomain.com, path=/                                                                                                                                             
debug: Verify: user=masau, groups=admins,server_admins,domain_admins,computer_admins                                                                                                       
error: Reply with error 401: TypeError: Cannot read property 'admins' of undefined                                                                                                            
    at /usr/lib/node_modules/authelia/dist/src/server/lib/access_control/AccessController.js:61:58                                                                                            
    at Array.reduce (<anonymous>)                                                                                                                                                             
    at AccessController.getMatchingGroupRules (/usr/lib/node_modules/authelia/dist/src/server/lib/access_control/AccessController.js:60:35)                                                   
    at AccessController.isAccessAllowed (/usr/lib/node_modules/authelia/dist/src/server/lib/access_control/AccessController.js:81:33)                                                         
    at /usr/lib/node_modules/authelia/dist/src/server/lib/routes/verify/get.js:28:44                                                                                                          
    at tryCatcher (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/util.js:16:23)                                                                                             
    at Promise._settlePromiseFromHandler (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/promise.js:512:31)                                                                  
    at Promise._settlePromise (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/promise.js:569:18)                                                                             
    at Promise._settlePromiseCtx (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/promise.js:606:10)                                                                          
    at Async._drainQueue (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/async.js:138:12)                                                                                    
    at Async._drainQueues (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/async.js:143:10)                                                                                   
    at Immediate.Async.drainQueues (/usr/lib/node_modules/authelia/node_modules/bluebird/js/release/async.js:17:14)                                                                           
    at runCallback (timers.js:781:20)
    at tryOnImmediate (timers.js:743:5)
    at processImmediate [as _immediateCallback] (timers.js:714:5)
info: Request GET handled on /?redirect=https%3A%2F%2Fauthelia.mydomain.com%2F
debug: First factor: headers are {"host":"authelia.mydomain.com:8080","user-agent":"Mozilla/5.0 (X11; Linux x86_64; rv:55.0) Gecko/20100101 Firefox/55.0","accept":"text/html,application/xhtml+
xml,application/xml;q=0.9,*/*;q=0.8","accept-language":"en-US,en;q=0.5","accept-encoding":"gzip, deflate","cookie":"_ga=GA1.2.1089724715.1482724523; connect.sid=s%3A2D893_odUMDjihoMoXsAhTEKM
vcHHtQl.Gm7tJIdU6b82wiTu1MMtpHTAtzbHm4Lrt5YuCED2hww","dnt":"1","connection":"keep-alive","upgrade-insecure-requests":"1","if-none-match":"W/\"635-CCpWJ/6kP59WBFIXOzmi385+dEc\""}
info: Request POST handled on /api/firstfactor?redirect=https://authelia.mydomain.com/&only_basic_auth=false
```

Nginx server config:
```
server {
  listen 443;
  root /srv/http;
  server_name authelia.mydomain.com;

  ssl on;
  ssl_certificate /etc/nginx/certs/authelia.mydomain.com/fullchain.pem;
  ssl_certificate_key /etc/nginx/certs/authelia.mydomain.com/privkey.pem;
  ssl_session_cache shared:SSL:10m;
  client_max_body_size 1024m;

  location /auth_verify {
    internal;
    proxy_set_header  X-Original-URI $request_uri;
    proxy_set_header  X-Real-IP $remote_addr;
    proxy_set_header  Host $http_host;

    proxy_pass        http://authelia.mydomain.com:8080/verify;
  }

  location / {
    auth_request /auth_verify;

    auth_request_set $redirect $upstream_http_redirect;
    proxy_set_header Redirect $redirect;

    auth_request_set $user $upstream_http_remote_user;
    proxy_set_header X-Forwarded-User $user;

    auth_request_set $groups $upstream_http_remote_groups;
    proxy_set_header Remote-Groups $groups;

    error_page 401 =302 http://authelia.mydomain.com:8080?redirect=$redirect;
    error_page 403 = http://authelia.mydomain.com:8080/error/403;
  }
}
```

LDAP portion of authelia config:
```
# LDAP configuration
#
# Example: for user john, the DN will be cn=john,ou=users,dc=example,dc=com
ldap:
  # The url of the ldap server
  url: ldap://ipa.mydomain.com

  # The base dn for every entries
  base_dn: dc=mydomain,dc=com

  # An additional dn to define the scope to all users
  additional_users_dn: cn=users,cn=accounts

  # The users filter.
  # {0} is the matcher replaced by username.
  # 'cn={0}' by default.
  users_filter: uid={0}

  # An additional dn to define the scope of groups
  additional_groups_dn: cn=groups,cn=accounts

  # The groups filter.
  # {0} is the matcher replaced by user dn.
  # 'member={0}' by default.
  groups_filter: (&(member={0})(objectclass=groupofnames))

  # The attribute holding the name of the group
  group_name_attribute: cn

  # The attribute holding the mail address of the user
  mail_attribute: mail

  # The username and password of the admin user.
  user: uid=authelia_admin,cn=users,cn=accounts,dc=mydomain,dc=com
  password: "mypassword"
```

**Comment by masau** [comment]
Found my problem - I deleted the groups section of access_control in config.yml which caused 'that.configuration.groups' to be undefined. More or less the same thing happens if the users section is deleted.

https://github.com/clems4ever/authelia/blob/15374f39d1ffb5979b31eade455420958517a358/server/src/lib/access_control/AccessController.ts#L78

**Comment by clems4ever** [comment]
Hello @masau, thanks for the troubleshooting. Indeed, this access is not protected. I will provide a quick fix for it. Thank you so much.

**Comment by clems4ever** [comment]
I will make Authelia resilient if you remove those keys and I also had in mind to create a configuration checker executed at startup so that one can find any issue with configuration very early.

**Comment by clems4ever** [comment]
Hello @masau , it is now fixed with #127. Every key in access_control is now optional and replaced by a default value if not provided. That way, Authelia always have a complete configuration to work with.
